### PR TITLE
Migrate kubernetes_e2e timeouts to a --timeout flag

### PR DESCRIPTION
**Pull request by fejta**
Also update the unit test to ensure that we leave kubetest at least 20 minutes to timeout.
* A few jobs set `kubetest --timeout` > `bootstrap.py --timeout`
* A variety of jobs set `kubetest --timeout` == `bootstrap.py --timeout`

Both of these situations abort kubetest timeout behavior (which includes shutting down the cluster and collecting logs)

Note the modifications to migrate `KUBETEST_TIMEOUT` in `foo.env` to `--timeout` in `config.json[foo][args]` was done by `jobs/move_timeout.py`

/assign @ixdy @krzyzacy 
ref #2829

### CODE REVIEW

File: jobs/config.json [hunk 1]
```diff
@@ -71,7 +71,9 @@
       "--env-file=platforms/gce.env", 
       "--env-file=jobs/ci-kubernetes-charts-gce.env", 
       "--test=false", 
-      "--mount-paths=$GOPATH/src/k8s.io/charts:/src/k8s.io/charts"
+      "--mount-paths=$GOPATH/src/k8s.io/charts:/src/k8s.io/charts", 
+      "--timeout=180m", 
+      "--timeout=None"
```
Comment (krzyzacy) [inline jobs/config.json]: that None does not look too correct

Comment (fejta) [inline jobs/config.json]: Done

**Comment by krzyzacy** [comment]
enjoy rebase :-)

**Comment by fejta** [comment]
Rebased

File: scenarios/kubernetes_e2e.py [hunk 2]
```diff
@@ -234,6 +234,7 @@ def add_environment(self, *envs):
             'GOPATH',
             'GOROOT',
             'HOME',
+            'KUBEKINS_TIMEOUT',  # Use --timeout
```
Comment (krzyzacy) [inline scenarios/kubernetes_e2e.py]: this is docker-mode only. I would suggest the same as https://github.com/kubernetes/test-infra/pull/2833#discussion_r118083089

Comment (fejta) [inline scenarios/kubernetes_e2e.py]: Done

File: jenkins/bootstrap_test.py [hunk 3]
```diff
@@ -2006,17 +2044,18 @@ def testEnvsNoExport(self):
...
                     'E2E_UP=',
-                    'USE_KUBEMARK=',
-                    'CHARTS_TEST=',
-                    'PERF_TESTS=',
-                    'FEDERATION_UP=',
                     'FEDERATION_DOWN=',
-                    'JENKINS_FEDERATION_PREFIX='
+                    'FEDERATION_UP=',
+                    'JENKINS_FEDERATION_PREFIX=',
+                    'KUBEKINS_TIMEOUT=',
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: hummm, exclude kops ones somehow

Comment (fejta) [inline jenkins/bootstrap_test.py]: Why?

Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: they are still using KUBEKINS_TIMEOUT?

Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: hey yeah that looks better

File: jenkins/bootstrap_test.py [hunk 4]
```diff
@@ -1859,34 +1867,64 @@ def testValidTimeout(self):
...
                     mat = re.match(r'export KUBEKINS_TIMEOUT="(\d+)m".*', line)
-                else:
+                    self.assertTrue(mat, 'Bad KUBEKINS_TIMEOUT in %s' % job)
+                    kubetest_timeout = int(mat.group(1))
+            elif not modern:
+                realjob = self.GetRealBootstrapJob(job)
+                self.assertTrue(realjob)
+                self.assertIn('timeout', realjob, job)
+                container_timeout = int(realjob['timeout'])
+                self.assertIn('kops', job)  # TODO(fejta): migrate kops to use --timeout
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: I think this trunk can be gone now

Comment (fejta) [inline jenkins/bootstrap_test.py]: Done

File: jenkins/bootstrap_test.py (hunk 3 above)
Comment (fejta) [inline jenkins/bootstrap_test.py]: Done

File: verify/BUILD [hunk 5]
```diff
@@ -6,7 +6,6 @@ py_test(
     name = "verify-boilerplate",
     srcs = ["verify-boilerplate.py"],
     data = ["//:all-srcs"],
-    tags = ["manual"],
```
Comment (krzyzacy) [inline verify/BUILD]: a separate PR?

**Comment by krzyzacy** [comment]
/lgtm

The timeout part is great, move the logic https://github.com/kubernetes/test-infra/pull/2904 out before merge :-)

File: verify/BUILD (hunk 5 above)
Comment (fejta) [inline verify/BUILD]: Done

**Comment by fejta** [comment]
/fyi @spxtr @ixdy 
I'm going to merge this and then update the prow job config. Will try and scan for errors

**Pull request merged**